packages/react/select/src/Select.tsx

```diff
@@ -251,7 +251,7 @@ const SelectTrigger = React.forwardRef<SelectTriggerElement, SelectTriggerProps>
           data-state={context.open ? 'open' : 'closed'}
           disabled={isDisabled}
           data-disabled={isDisabled ? '' : undefined}
-          data-placeholder={context.value === undefined ? '' : undefined}
+          data-placeholder={shouldShowPlaceholder(context.value) ? '' : undefined}
```

Comment:
This is combination with the next change below (`placeholder = ''`) means users will always be able to style differently, even if they don't want any visible placeholder text.